front/src/modules/powerRestriction/helpers/createPathStep.ts

```diff
@@ -18,25 +18,28 @@ const findTrackSectionIndex = (position: number, tracksLengthCumulativeSums: num
   return -1;
 };
 
-const findTrackSection = (
-  position: number, // in mm
-  tracksLengthCumulativeSums: number[],
-  pathProperties: ManageTrainSchedulePathProperties
-) => {
-  const index = findTrackSectionIndex(position, tracksLengthCumulativeSums);
-  return index !== -1 ? pathProperties.trackSectionRanges[index] : null;
-};
-
-const calculateOffset = (
-  trackSectionRange: TrackRange,
-  position: number, // in mm
-  tracksLengthCumulativeSums: number[] // in meters
+/**
+ * Return the track offset corresponding to the given position on path, composed of the track section id
+ * and the offet on this track section in mm
+ */
+export const findTrackSectionOffset = (
+  positionOnPath: number, // in mm
+  tracksLengthCumulativeSums: number[], // in mm
+  trackRangesOnPath: TrackRange[]
 ) => {
-  const index = findTrackSectionIndex(position, tracksLengthCumulativeSums);
-  const inferiorSum = tracksLengthCumulativeSums[index];
-  return trackSectionRange.direction === 'START_TO_STOP'
-    ? inferiorSum - position
-    : position - inferiorSum;
+  const index = findTrackSectionIndex(positionOnPath, tracksLengthCumulativeSums);
+  const trackRange = trackRangesOnPath[index];
+  if (!trackRange) return null;
+
+  // compute offset
+  const inferiorSum = index > 0 ? tracksLengthCumulativeSums[index - 1] : 0;
+  const offsetOnTrackRange = positionOnPath - inferiorSum;
+  const offsetOnTrackSection =
+    trackRange.direction === 'START_TO_STOP'
```

Comment:
Is there is several direction it could be nice if we have an enum

Reply:
This field is auto-generated from OpenAPI, so we can't manually define an enum. (I haven't looked whether it's possible to change the Rust counterpart to get an enum on the TypeScript side.)